components/script/dom/document.rs

```diff
@@ -997,6 +1002,27 @@ impl<'a> DocumentHelpers<'a> for &'a Document {
             .filter_map(HTMLIFrameElementCast::to_root)
             .find(|node| node.r().subpage_id() == Some(subpage_id))
     }
+
+    /// Find and cache the first title element in the document.
+    fn update_title_element(self) {
+        let title = self.GetDocumentElement().and_then(|root| {
+            if root.r().namespace() == &ns!(SVG) && root.r().local_name() == &atom!("svg") {
+                // Step 1.
+                NodeCast::from_ref(root.r()).child_elements().find(|node| {
+                    node.r().namespace() == &ns!(SVG) &&
+                    node.r().local_name() == &atom!("title")
+                }).map(NodeCast::from_root)
+            } else {
+                // Step 2.
+                NodeCast::from_ref(root.r())
+                         .traverse_preorder()
+                         .find(|node| node.r().is_htmltitleelement())
+            }
+        });
+
+        self.current_title_element.set(
+            title.r().and_then(HTMLTitleElementCast::to_ref).map(JS::from_ref));
```

Comment:
AFAICT, there should be SVG elements that aren't deriving from HTMLElement, but that's a bit huge to implement right now just for this. Maybe cache the title only if not in a HTML document for now?